cli/azd/pkg/project/container_helper.go

```diff
@@ -39,15 +39,33 @@ func NewContainerHelper(
 	}
 }
 
-func (ch *ContainerHelper) RegistryName(ctx context.Context) (string, error) {
-	loginServer, has := ch.env.LookupEnv(environment.ContainerRegistryEndpointEnvVarName)
-	if !has {
+// RegistryName returns the name of the container registry to use for the current environment from the following:
+// 1. AZURE_CONTAINER_REGISTRY_ENDPOINT environment variable
+// 2. docker.registry from the service configuration
+func (ch *ContainerHelper) RegistryName(ctx context.Context, serviceConfig *ServiceConfig) (string, error) {
+	var loginServer string
+	envVarRegistryName, _ := ch.env.LookupEnv(environment.ContainerRegistryEndpointEnvVarName)
+	yamlRegistryName, _ := serviceConfig.Docker.Registry.Envsubst(ch.env.Getenv)
```

Comment:
do not ignore errors. Use log.prinln or something similar where debug logs can show the error while trying to fetch the value